libraries/botbuilder-ai/src/languageTranslator.ts

```diff
@@ -46,37 +49,42 @@ export class LanguageTranslator implements Middleware {
                 return Promise.resolve();
             }
         }
-        // translate to bots language
-        return this.translateMessageAsync(context)
-        .then(() => next());
-        
-    }
-
-    /// Translate .Text field of a message, regardless of direction
-    private async translateMessageAsync(context: TurnContext): Promise<TranslationResult[]> {
-        
-
         // determine the language we are using for this conversation
         let sourceLanguage: string;
         if (this.getUserLanguage != undefined) {
             sourceLanguage = this.getUserLanguage(context);
-        } else if (context.activity.locale != undefined) {
-            sourceLanguage = context.activity.locale;
         } else {
             sourceLanguage = await this.translator.detect(context.activity.text);
         }
-
         
         let targetLanguage = (this.nativeLanguages.indexOf(sourceLanguage) >= 0) ? sourceLanguage : this.nativeLanguages[0];
+        
+        await this.translateMessageAsync(context, context.activity, sourceLanguage, targetLanguage);
+
+        if (this.translateBackToUserLanguage) {
+            context.onSendActivities(async (newContext, activities, newNext) => {
+                let currentMessageActivity = activities[0];
+                await this.translateMessageAsync(newContext, currentMessageActivity, targetLanguage, sourceLanguage);
+                activities[0].text = currentMessageActivity.text;
```

Comment:
>activities[0].text = currentMessageActivity.text; [](start = 16, length = 49)

This assignment happens translateMessageAsync(). You're not cloning the activity so why the additional assignment?